middleware/minimal.go

```diff
@@ -0,0 +1,45 @@
+package middleware
+
+import (
+	"net/http"
+)
+
+type minimalWriter struct {
+	http.ResponseWriter
+	status int
+}
+
+func (w *minimalWriter) Write(data []byte) (int, error) {
+	if w.status == 0 {
+		w.WriteHeader(http.StatusOK)
+	}
+
+	if w.status == http.StatusNoContent {
+		return 0, nil
+	}
+
+	return w.ResponseWriter.Write(data)
+}
+
+func (w *minimalWriter) WriteHeader(statusCode int) {
+	if statusCode >= 200 && statusCode < 300 {
+		statusCode = http.StatusNoContent
```

Comment:
https://tools.ietf.org/html/rfc7240#section-4.2 has an example case where a 201 Created is an appropriate status code.